drivers/flash/soc_flash_mcux.c

```diff
@@ -172,6 +172,20 @@ static int flash_mcux_erase(const struct device *dev, off_t offset,
 			(FMU_Type *) DT_INST_REG_ADDR(0),
 #endif
 			addr, len, kFLASH_ApiEraseKey);
+
+#ifdef CONFIG_SOC_SERIES_MCXW
+	/* clear the flash cache */
+	const uint32_t mcxw_flash_cache_clear_bit = BIT(8);
+	const uint32_t mcxw_smscm_ocmdr0_reg_addr = 0x40015400;
+	volatile uint32_t *smscm_ocmdr0 = (volatile uint32_t *)mcxw_smscm_ocmdr0_reg_addr;
+	*smscm_ocmdr0 |= mcxw_flash_cache_clear_bit;
+	/* clear code cache */
+	const uint32_t mcxw_code_cache_clear_bit = BIT(0);
+	const uint32_t mcxw_mcm_cpcr2_reg_addr = 0xe0080034;
+	volatile uint32_t *mcm_cpcr2 = (volatile uint32_t *)mcxw_mcm_cpcr2_reg_addr;
+	*mcm_cpcr2 |= mcxw_code_cache_clear_bit;
+#endif
```

Comment:
Would you not need similar cache management code after writes? 
Also is it possible to invalidate a certain range instead of the whole cache?

Reply:
doesn't seem to be needed after writes since write goes through the cache, erase operation doesn't involve writing any data

and no, there is no way to clear a certain range as far as I can find

Reply:
i updated the PR to also put it after programs